addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4672,6 +4688,155 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
         // to be refactored when the new editor is merged
         this.$target.dblclick();
     },
+    /**
+     * Adds an anchor to the url.
+     * Here "anchor" means a specific section of a page.
+     *
+     * @see this.selectClass for parameters
+     */
+    setAnchor(previewMode, value) {
+        const linkEl = this.$target[0].parentElement;
+        let url = linkEl.getAttribute('href');
+        if (url.includes('#')) {
+            url = url.split('#')[0];
+        }
+        linkEl.setAttribute('href', url + value);
+    },
+    /**
+     * Makes the image a clickable link by wrapping it in an <a>.
+     * This function is also called for the opposite operation.
+     *
+     * @see this.selectClass for parameters
+     */
+    setLink() {
+        let linkEl = this.$target[0].parentElement;
+        const parentEl = this.$target[0].parentNode;
+        if (linkEl.tagName !== 'A') {
+            // Wrap this.$target[0] like this: <a>this.$target[0]</a>.
+            linkEl = document.createElement('a');
+            parentEl.replaceChild(linkEl, this.$target[0]);
+            linkEl.appendChild(this.$target[0]);
+        } else {
+            // Remove the parent anchor.
+            parentEl.replaceWith(this.$target[0]);
+        }
+    },
+    /**
+     * Makes that when the link (parent element of the image) is clicked, the
+     * url will open on another page. This function is also called for the
+     * opposite operation (open the url in the current browser page).
+     *
+     * @see this.selectClass for parameters
+     */
+    setNewWindow() {
+        const linkEl = this.$target[0].parentElement;
+        if (linkEl.getAttribute('target') === '_blank') {
+            return linkEl.removeAttribute('target');
+        }
+        linkEl.setAttribute('target', '_blank');
+    },
+    /**
+     * Records the target url of the hyperlink.
+     *
+     * @see this.selectClass for parameters
+     */
+    setUrl(previewMode, url) {
+        const linkEl = this.$target[0].parentElement;
+        if (linkEl.tagName !== 'A') {
+            return;
+        }
```

Comment:
Indeed, cases where this happens are rare... But they do exist, it happens ~ one time out of 100 while changing the url and clicking on the button to remove the link or by clicking several times on the button rather quickly.

Reply:
I don't understand how that could be possible, if that is the case, the cause has to be fixed instead of shielding code against it like this :+1:

Reply:
I investigated the code to understand how it is possible to have this function (setUrl) called when the parent element A has already been deleted by my other function (setLink). In fact in the _onInputBlur function of the InputUserValueWidget, the code is in a setTimeOut. This makes it possible to have the function that deletes the A before the setUrl function if both events are triggered in a very short period of time. I don't see an ideal solution... I could put the _onButtonClick code in a setTimeOut as well but I doubt that would be a better solution....

Reply:
Mmh ok, good catch. Maybe we should investigate if that `setTimeout` is still necessary with the new editor. Maybe we can remove it in a commit prior to your commits here and actually use the "change" event instead of the blur + keydown-for-enter implementation.

If not possible, then we'll keep this `if` with a comment explaining why

Reply:
Indeed, this `setTimeout` seems to be useless now. So I removed it. On the other hand, passing from the blur event to change seems to me a bad idea. For the moment, `_onUserValueChange` is not called if the user is changing the value of we-input and changes the tab on his browser. If we want to keep this behaviour, using the change event could cause the editor to be in an inconsistent state.
Ex: 
- User changes border value to 10px
- He changes the tab (the change event is called but not `_onUserValueChange`)
- He returns to the tab and blurs the border input.
- The change event is not called and so `_onUserValueChange` is not called either.
=> We end up with the editor having 10px for the border value while the dom is still 0.

Reply:
> He changes the tab (the change event is called but not _onUserValueChange)

Why not? :thinking:

Reply:
Because in the _onInputBlur function there is this :
```
if (ev.currentTarget === document.activeElement) {
    return;
}
```
And as you can see here https://jsfiddle.net/gdi_odoo/15mt4h3o/7/ when you focus an input then change tab, the blur event will be trigger but the input is still the activeElement.

Reply:
Well yes... but why would you keep that code if you remove the setTimeout and change the blur event to a change event ? :thinking:

Reply:
In order to keep the same behaviour.
I can remove this code and use the change event but this implies that when the user changes tabs, the last changes will be applied (whereas they were not applied before).
I don't know if this behaviour is important or not. If it is not, no problem to remove it. :)

Reply:
That line was there for the reason explained just above it; the behavior with the tab change is just a side effect and really I don't think anyone would find that important to preserve :grimacing: Especially since using the `change` event, especially without that strange if... is just the normal way to code the input change listening so all other behavior in Odoo and other software/website would have that behavior :wink: 

So yes, if you confirmed that this code is not needed anymore to counter the "bug" that is explained above the `setTimeout`, change the whole thing to use the `change` event in a dedicated commit to start this PR, that will normally allow you:
- To remove the `_onInputBlur` code entirely
- To remove the "Enter" key handling

The less code the better :+1: The less hacks the better :+1: